public/js/src/module/controller-webform.js

```diff
@@ -133,6 +155,16 @@ function _checkAutoSavedRecord() {
         } );
 }
 
+/**
+ * @param {string} enketoId
+ */
+function _updateFormData( enketoId ) {
+    return formCache.get( { enketoId } )
+        .then( survey => {
+            formData.external = survey.externalData;
+        } );
+}
+
```

Comment:
I really did not want to add more logic in this module without tests, but this was the only way I could find to ensure last-saved secondary instance references would update on save/form reset.